### PR TITLE
refactor: migrate GET /adminform endpoint to Typescript

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds a new `admin-form` module, and refactors the GET /adminform endpoint to the new layered architecture.

## Solution

**Features**:

- add `FormModel#getDashboardForms` static function
- add `FormService#getDashboardForms` service function
- add `AdminFormController#handleListDashboardForms` handler for GET /adminforms endpoint

**Improvements**:

- Use new controller handler in admin-forms.server.routes.js and remove old `list` function.

## Tests

- add tests for getDashboardForms service function
- add tests for getDashboardForms model static function

### CODE REVIEW

File: src/app/modules/form/admin-form/admin-form.controller.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  const dashboardResult = await getDashboardForms(req.session.user._id)
+
+  if (dashboardResult.isErr()) {
+    const { error } = dashboardResult
+    logger.error({
+      message: 'Error listing dashboard forms',
+      meta: {
+        action: 'handleListDashboardForms',
```
Comment (mantariksh) [inline src/app/modules/form/admin-form/admin-form.controller.ts:33]: might be good to log the user ID as well

File: tests/unit/backend/models/form.server.model.spec.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
-import { ObjectID } from 'bson'
-import { merge, omit } from 'lodash'
+import { ObjectId } from 'bson-ext'
```
Comment (mantariksh) [inline tests/unit/backend/models/form.server.model.spec.ts:1]: out of curiosity, why did you change it from `bson` to `bson-ext`?

File: tests/unit/backend/models/form.server.model.spec.ts [hunk 3]
```diff
@@ -872,13 +870,105 @@ describe('Form Model', () => {
...
+      })
+      // Populate multiple forms with different permissions.
+      // Is admin.
+      const userOwnedForm = await Form.create(MOCK_EMAIL_FORM_PARAMS)
+      // Has permissions.
+      const userPermissionForm = await Form.create({
+        ...MOCK_ENCRYPTED_FORM_PARAMS,
+        admin: diffPopulatedAdmin._id,
+        permissionList: [{ email: populatedAdmin.email, write: true }],
+      })
```
Comment (mantariksh) [inline tests/unit/backend/models/form.server.model.spec.ts:916]: lol, I wonder if it might be better to separate these tests, eg one test each for admin, write permissions, read permissions, archived, no permissions

File: tests/unit/backend/models/form.server.model.spec.ts [hunk 4]
```diff
@@ -872,13 +870,105 @@ describe('Form Model', () => {
...
+          // Should return form where admin has permission.
+          merge(
+            pick(userPermissionForm.toObject(), [
+              '_id',
+              'title',
+              'lastModified',
+              'status',
+              'form_fields',
+              'responseMode',
+            ]),
```
Comment (mantariksh) [inline tests/unit/backend/models/form.server.model.spec.ts:960]: can extract this array to make the test more readable

File: src/app/models/form.server.model.ts [hunk 5]
```diff
@@ -501,6 +495,32 @@ const compileFormModel = (db: Mongoose): IFormModel => {
...
+  ): Promise<DashboardFormView[]> {
+    return (
+      this.find()
+        // List forms when either the user is an admin or collaborator.
+        .or([{ 'permissionList.email': userEmail }, { admin: userId }])
+        // Filter out archived forms.
+        .where('status')
+        .ne(Status.Archived)
+        // Project selected fields.
+        .select('_id title admin lastModified status form_fields')
```
Comment (mantariksh) [inline src/app/models/form.server.model.ts:511]: might be cleaner to extract this into a constants file somewhere

**Review by mantariksh — APPROVED** [review]
minor suggestions!

**Review by liangyuanruo — APPROVED** [review]

File: tests/unit/backend/models/form.server.model.spec.ts (hunk 2 above)
Comment (karrui) [reply tests/unit/backend/models/form.server.model.spec.ts:1]: no reason tbh, making it consistent with our usages somewhere else I guess. Its all over the place right now (of my own doing, sorry)

File: tests/unit/backend/models/form.server.model.spec.ts (hunk 3 above)
Comment (karrui) [reply tests/unit/backend/models/form.server.model.spec.ts:916]: eh, this is a base function that is supposed to return all the above types. so keeping it all in one test works too since it will fail if any of the types does not get returned. 

building on your comment, will make the test more inclusive, add write permissions test form too just in case that fails

File: src/app/models/form.server.model.ts (hunk 5 above)
Comment (karrui) [reply src/app/models/form.server.model.ts:511]: Only used here, no need to extract for now

File: tests/unit/backend/models/form.server.model.spec.ts (hunk 3 above)
Comment (mantariksh) [reply tests/unit/backend/models/form.server.model.spec.ts:916]: I know it works, I'm just trying to improve the readability and modularity of the tests
